lib/checkbufferoverrun.cpp

```diff
@@ -881,3 +881,47 @@ bool CheckBufferOverrun::analyseWholeProgram1(const CTU::FileInfo *ctu, const st
     return true;
 }
 
+void CheckBufferOverrun::objectIndex()
+{
+    const SymbolDatabase *symbolDatabase = mTokenizer->getSymbolDatabase();
+    for (const Scope *functionScope : symbolDatabase->functionScopes) {
+        for (const Token *tok = functionScope->bodyStart; tok != functionScope->bodyEnd; tok = tok->next()) {
+            if (!Token::Match(tok, "["))
+                continue;
+            const Token *obj = tok->astOperand1();
+            const Token *idx = tok->astOperand2();
+            if (!idx || !obj)
+                continue;
+            if (!idx->hasKnownIntValue())
```

Comment:
what do you think about writing something like:

    if (idx->hasKnownIntValue() && idx->getKnownIntValue() == 0)
        continue;

And then write something like:

    array index out of bounds. the array '&x' has size 1. index is/might be non-zero.

That could cause more warnings.. do you think it would be noise?

Reply:
That might be good, but we should probably make it a warning or style when the value is not known.

Reply:
yes.. `warning` seems appropriate.. it should be used when there is possible UB.

Reply:
So you made the right choice already.